sympy/polys/tests/test_hypothesis.py

```diff
@@ -1,15 +1,27 @@
-from hypothesis import given
+from hypothesis import given, assume
 from hypothesis import strategies as st
 from sympy.abc import x
 from sympy.polys.polytools import Poly
+from sympy.polys.domains import ZZ
+
+
+@st.composite
+def coefficients(draw: st.DrawFn, empty=True):
+    l = [ZZ(i) for i in draw(st.lists(st.integers()))]
+    if not empty:
+        assume(len(l) > 0)
```

Comment:
Try to avoid assuming when you can construct the strategy in a way that never meets the filter condition.
```suggestion
    min_size = 0 if empty else 1
    l = [ZZ(i) for i in draw(st.lists(st.integers(), min_size=min_size))]
```

nit: I think its clearer to keep one piece of logic to one line, e.g.

```suggestion
    min_size = 0 if empty else 1
    raw_l = draw(st.lists(st.integers(), min_size=min_size))
    l = [ZZ(i) for i in l]
```